### PR TITLE
Guidance on contribution content and structure

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
Give some guidance on where to contribute, what content is expected, and how content should be structured.

This complements the existing contribution guidance: CONTRIBUTING.md in the mbedtls repository, and the guide to the review process from a contributor's point of view. It might make sense to combine this information in a single document, but that will be for later.

### CODE REVIEW

File: kb/development/contribution-structure.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+* We don't have the capacity to test on many platforms, so we don't expect tests for issues that are specific to a particular CPU or operating system. If you're submitting a fix for a bug on a platform that we don't have, please test locally.
+* Mbed TLS is very configurable. We test a lot of different configurations, but we often won't add testing for build failures that only happen in relatively rare configurations.
+* There are certain errors that we cannot trigger on purpose at the moment: hardware accelerator errors, and memory allocation failures. It is generally acceptable to omit a non-regression test when fixing a bug in the recovery from such an error.
+
+#### Running tests locally
+
+For basic unit tests, you can just compile the project and run `make test`. Note that this only tests the default configuration. The entry point to testing with many different configurations is [`tests/scripts/all.sh`](https://github.com/Mbed-TLS/mbedtls/blob/development/tests/scripts/all.sh).
+
+To run some tests, in particular TLS testing, you will need additional tooling. See the [Testing and CI guide](/kb/testing/testing-ci.md) for more information.
```
Comment (eleuzi01) [inline kb/development/contribution-structure.md:59]:
```suggestion
To run some tests, in particular TLS testing, you will need additional tooling. See the [Testing and CI guide](../../kb/testing/testing-ci.md) for more information.
```

**Review by eleuzi01 — COMMENTED** [review]
I think it's a very useful doc, thanks! Just a couple minor comments

**Review by eleuzi01 — APPROVED** [review]
LGTM, thanks

**Review by davidhorstmann-arm — APPROVED** [review]
LGTM, thanks!

**Pull request merged**